pypika/queries.py

```diff
@@ -2042,6 +2058,77 @@ def __repr__(self) -> str:
         return self.__str__()
 
 
+class CreateIndexBuilder:
+
+    def __init__(self, dialect: Optional[Dialects] = None) -> None:
+        self._dialect = dialect
+        self._index = None
+        self._columns = []
+        self._table = None
+        self._wheres = None
+        self._is_unique = False
+        self._if_not_exists = False
+
+    @builder
+    def create_index(self, index: Union[str, Index]) -> "CreateIndexBuilder":
+        self._index = index
+
+    @builder
+    def columns(self, *columns: Union[str, TypedTuple[str, str], Column]) -> "CreateIndexBuilder":
+        for column in columns:
+            if isinstance(column, str):
+                column = Column(column)
+            elif isinstance(column, tuple):
+                column = Column(column_name=column[0], column_type=column[1])
+            self._columns.append(column)
+
+    @builder
+    def on(self, table: Union[Table, str]) -> "CreateIndexBuilder":
+        self._table = table
+
+    @builder
+    def where(self, criterion: Union[Term, EmptyCriterion]) -> "CreateIndexBuilder":
+        """
+        Partial index where clause.
+        """
+        if isinstance(criterion, EmptyCriterion):
+            return self
+
+        if self._wheres:
+            self._wheres &= criterion
+        else:
+            self._wheres = criterion
+        return self
```

Comment:
I think that using the `@builder` annotation we do not need to return `self` because it does it for you

Reply:
Yeah I agree; my pycharm linter (sonarlint) was moaning that return type hint was there but no return. I'll remove for consistency with the rest of the code.